### ISSUE TITLE
removing links and deleting LogicNodes causes exceptions

### ISSUE THREAD

**Issue by martin-hackenberg-paradoxcat** [GENIVI/ramses-logic#20]
Occurred in ramses-logic version: 0.3
Tests to reproduce issue:
```
TEST(ramses_logic, unlinkAndDestroy_nodeBinding) {
	rlogic::LogicEngine logicEngine;
	ramses::RamsesFramework ramsesFramework;
	ramses::RamsesClient& client = *ramsesFramework.createClient("example client");
	ramses::Scene* scene = client.createScene(ramses::sceneId_t(123u), ramses::SceneConfig(), "some scene");
	ramses::Node* node = scene->createNode("some node");

	auto* outScript = logicEngine.createLuaScriptFromSource(R"(
function interface()
	IN.in_float = FLOAT
    OUT.out_vec3f = VEC3F
end

function run()
	OUT.out_vec3f = { IN.in_float, 0.0, 0.0 }
end
)");

	auto* binding = logicEngine.createRamsesNodeBinding("some node binding");
	binding->setRamsesNode(node);

	const rlogic::Property* nodeTranslation {nullptr};
	for (size_t i{0}; i < binding->getInputs()->getChildCount(); i++) {
		if (binding->getInputs()->getChild(i)->getName() == "translation") {
			nodeTranslation = binding->getInputs()->getChild(i);
		}
	}
	ASSERT_TRUE(nodeTranslation != nullptr);
	const rlogic::Property* scriptOutVec3f {nullptr};
	for (size_t i {0}; i < outScript->getOutputs()->getChildCount(); i++) {
		if (outScript->getOutputs()->getChild(i)->getName() == "out_vec3f") {
			scriptOutVec3f = outScript->getOutputs()->getChild(i);
		}
	}
	ASSERT_TRUE(scriptOutVec3f != nullptr);

	ASSERT_TRUE(logicEngine.link(*scriptOutVec3f, *nodeTranslation));
	ASSERT_TRUE(logicEngine.unlink(*scriptOutVec3f, *nodeTranslation));
	ASSERT_TRUE(logicEngine.destroy(*binding));
	ASSERT_TRUE(logicEngine.update());
}
```

```
TEST(ramses_logic, unlinkAndDestroyScript_multipleLinks) {
	rlogic::LogicEngine logicEngine;
	ramses::RamsesFramework ramsesFramework;
	ramses::RamsesClient& client = *ramsesFramework.createClient("example client");
	ramses::Scene* scene = client.createScene(ramses::sceneId_t(123u), ramses::SceneConfig(), "some scene");
	ramses::Node* node = scene->createNode("some node");

	auto* outScript = logicEngine.createLuaScriptFromSource(R"(
function interface()
    OUT.translation = VEC3F
    OUT.visibility = BOOL
end

function run()
end
)");

	auto* binding = logicEngine.createRamsesNodeBinding("some node binding");
	binding->setRamsesNode(node);

	const rlogic::Property* nodeTranslation {nullptr};
	const rlogic::Property* nodeVisibility {nullptr};
	for (size_t i{0}; i < binding->getInputs()->getChildCount(); i++) {
		if (binding->getInputs()->getChild(i)->getName() == "translation") {
			nodeTranslation = binding->getInputs()->getChild(i);
		}
		if (binding->getInputs()->getChild(i)->getName() == "visibility") {
			nodeVisibility = binding->getInputs()->getChild(i);
		}
	}
	ASSERT_TRUE(nodeTranslation != nullptr);
	ASSERT_TRUE(nodeVisibility != nullptr);

	const rlogic::Property* scriptTranslation {nullptr};
	const rlogic::Property* scriptVisiblity {nullptr};
	for (size_t i {0}; i < outScript->getOutputs()->getChildCount(); i++) {
		if (outScript->getOutputs()->getChild(i)->getName() == "translation") {
			scriptTranslation = outScript->getOutputs()->getChild(i);
		}
		if (outScript->getOutputs()->getChild(i)->getName() == "visibility") {
			scriptVisiblity = outScript->getOutputs()->getChild(i);
		}
	}
	ASSERT_TRUE(scriptTranslation != nullptr);
	ASSERT_TRUE(scriptVisiblity != nullptr);

	ASSERT_TRUE(logicEngine.link(*scriptVisiblity, *nodeVisibility));
	ASSERT_TRUE(logicEngine.link(*scriptTranslation, *nodeTranslation));

	ASSERT_TRUE(logicEngine.update());

	ASSERT_TRUE(logicEngine.unlink(*scriptVisiblity, *nodeVisibility));
	ASSERT_TRUE(logicEngine.unlink(*scriptTranslation, *nodeTranslation));
	ASSERT_TRUE(logicEngine.destroy(*outScript));

	ASSERT_TRUE(logicEngine.update());
}
```

**Comment by violinyanev** [comment]
Looking into it

**Comment by violinyanev** [comment]
Confirmed - it's a bug...
Thanks for the nicely prepared test - took me 2 min to analyze it with it! ;) Fix incoming!

**Comment by violinyanev** [comment]
Fix shipped with v0.3.1
